### ISSUE TITLE
Form Block: Make It Easier To Select Parent Block

### ISSUE THREAD

**Issue by donalirl** [Automattic/jetpack#38368]
### Impacted plugin

Jetpack

### What

In the settings of the individual form field blocks, let's add a button to navigate to the overall Form block settings.

### How

A common user pain point is trying to navigate back to the parent Form block to edit overall form settings like the submission and email. It's a pain point because when you click on the form block, it selects an individual form field like Name, Message, etc:

https://github.com/user-attachments/assets/872eb36a-5fce-4277-9e74-355f6b2e1f25

The methods to navigate to the parent block are not intuitive for many users:

* click on the small space between two form fields
* use the parent block selector in the toolbar
* use list view

Those are not intuitive because when you click on the form, you expect to see the form settings, not different settings for the individual fields.

In #8491893-zen you can find an example of how the bot does a good job of guiding the user to the parent level Submission Settings, but the user can't find it because they're selecting the form field. To them, it seems like the form settings are missing but it's because they've selected an individual field.

One potential solution would be to add a button to each individual form field that takes the user to the parent block settings:
![Screenshot 2024-07-17 at 1 43 15 PM](https://github.com/user-attachments/assets/b8a88667-9ecd-44a3-bd2a-9fcefcddac5b)

**Comment by jeherve** [comment]
I believe that is something that may be better addressed in core directly, so it can benefit all blocks at once. There are a few ongoing discussions about better navigation between parent and inner blocks:

https://github.com/WordPress/gutenberg/issues/41924
https://github.com/WordPress/gutenberg/issues/46040

**Comment by simison** [comment]
I agree this is a generic issue that needs improving in Gutenberg, but I'd like to track this in case there's something we can improve regardless. E.g. make it less important to select parent forms block.

**Comment by simison** [comment]
Internal ref, Slack convo p1736516017313059-slack-C45SNKV4Z

Feedback was that design/Write modes address the issue, adding this to the board:
- https://github.com/Automattic/jetpack/issues/40955

**Comment by simison** [comment]
Matías noted elsewhere (internal ref p1HpG7-l0G-p2):

> The “form” should be a controlled block, meaning you get to access the children from the parent’s sidebar, and have easy built-in travel back to the wrapper.
